Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -476,12 +492,145 @@ void MemoryWidget::OnSearchAddress()
   m_search_offset->setPalette(offset_palette);
 }
 
-void MemoryWidget::ValidateSearchValue()
+void MemoryWidget::ValidateAndPreviewInputValue()
 {
+  m_data_preview->clear();
+  QString input_text = m_data_edit->text();
+  const auto combo_id = static_cast<InputId>(m_input_combo->currentData().toInt());
+
+  m_base_check->setEnabled(combo_id == InputId::U32 || combo_id == InputId::U16 ||
+                           combo_id == InputId::U8);
+
+  if (input_text.isEmpty())
+    return;
+
+  // Remove any spaces
+  if (combo_id != InputId::ASCII)
+    input_text.remove(QLatin1Char(' '));
+
   QFont font;
   QPalette palette;
+  QString hex_string;
+  bool good = true;
+
+  if (m_base_check->isChecked() && m_base_check->isEnabled())
+    input_text = QStringLiteral("0x") + input_text;
```

Comment:
I think you should use the radix and pass it as parameter to QString's conversion methods rather than prepending the hex prefix to the string.
```c++
const int radix = (m_base_check->isChecked() && m_base_check->isEnabled()) ? 16 : 0;
```